libs/martech/martech.js

```diff
@@ -97,49 +98,91 @@ const getDtmLib = (env) => ({
       : env.consumer?.marTechUrl || 'https://assets.adobedtm.com/d4d114c60e50/a0e989131fd5/launch-a27b33fc2dc0-development.min.js',
 });
 
-export default async function init({ persEnabled = false, persManifests }) {
+const setupEntitlementCallback = () => {
+  const setEntitlements = async (destinations) => {
+    const { default: parseEntitlements } = await import('../features/personalization/entitlements.js');
+    return parseEntitlements(destinations);
+  };
+
+  const getEntitlements = () => new Promise((resolve) => {
+    const handleEntitlements = (e) => {
+      window.removeEventListener(ALLOY_SEND_EVENT, handleEntitlements);
+
+      if (e.detail?.result?.destinations?.length) {
+        resolve(setEntitlements(e.detail.result.destinations));
+      } else {
+        resolve([]);
+      }
+    };
+    window.addEventListener(ALLOY_SEND_EVENT, handleEntitlements);
```

Comment:
Since there are multiple interact calls, are there also multiple alloy event calls or is it enough to just wait for the first one?

// @narcis-radu didn't you just have something similar with alloy/gnav tests on AEM?

Reply:
From my understanding, the first time this event fires that call will contain the XLG data.

Reply:
We should add some guardrails to ensure that's the case. We use a similar approach in FEDS and have to check against a parameter that has been used to perform the call.

Reply:
Can you send me more info on this?